internal/build_defs.bzl

```diff
@@ -184,3 +184,45 @@ def tsc_library(**kwargs):
       compiler = "//internal/tsc_wrapped:tsc",
       node_modules = "@build_bazel_rules_typescript_deps//:node_modules",
       **kwargs)
+
+# ******************* #
+# closure_ts_binary   #
+# ******************* #
+def _collect_es6_sources_impl(ctx):
+  """Rule which wraps the collect_es6_sources action for rules_closure.
+
+  Args:
+    ctx: the context.
+
+  Returns:
+    A closure_js_library with the rerooted files.
+  """
+  collected_es6_sources = collect_es6_sources(ctx)
+
+  js_module_roots = depset()
+  for prod_file in collected_es6_sources:
+    if "node_modules/" in prod_file.dirname:
+      js_module_roots += [prod_file.dirname[:prod_file.dirname.find('node_modules/')]]
+
+  return struct(
+    files = collected_es6_sources,
+    closure_js_library = struct(
+      srcs = collected_es6_sources,
+      js_module_roots = js_module_roots,
+    )
+  )
+
+_collect_es6_sources = rule(
+    attrs = {"deps": attr.label_list(mandatory = True)},
+    implementation = _collect_es6_sources_impl,
+)
+
+def closure_ts_binary(name, deps, **kwargs):
```

Comment:
I think we discussed that I'd rather let users see the `closure_js_binary` in its full glory, and direct their questions and wrath to owners of those rules.
I think it might be better for users to just expose the collect_sources as a closure_js_library dropin replacement:
```
ts_library(name = foo)
# This rule converts its deps to act like a `closure_js_library`
closure_ts_library(name = "foo_closure", deps = [":foo"])

closure_js_binary(deps = [":foo_closure"])
```

it means one more rule in users BUILD files but I think it's better to expose the fundamental building blocks rather than introduce a leaky abstraction.